src/btree/bt_read.c

```diff
@@ -528,9 +556,12 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
 	 * Don't free WT_REF.page_las, there may be concurrent readers.
 	 */
 	if (final_state == WT_REF_MEM && ref->page_las != NULL &&
-	    (!ref->page_las->skew_newest || ref->page_las->has_prepares))
+	    (!ref->page_las->skew_newest || ref->page_las->has_prepares)) {
 		WT_ERR(__wt_las_remove_block(
 		    session, ref->page_las->las_pageid));
+		__wt_free(session, ref->page_las->birthmarks);
+		__wt_free(session, ref->page_las);
```

Comment:
This is not being done at the moment in develop. What happens is if this page gets evicted again and we keep this lookside reference, when the page gets re-instantiated lookaside search is done to locate records with this las_pageid. No record is found because this block got removed. But searching for the record is an inefficiency we could fix.

Reply:
Removing eviction_to_lookaside resolved this issue in this branch. Also, I couldn't reproduce the mentioned case in develop. So now this comment can be ignored.